web/modules/custom/dpl_redia_legacy/src/Controller/RssFeeds/EventsController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Drupal\dpl_redia_legacy\Controller\RssFeeds;
+
+use Drupal\Core\Cache\CacheableResponse;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\File\FileUrlGeneratorInterface;
+use Drupal\Core\Url;
+use Drupal\dpl_redia_legacy\RediaEvent;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use function Safe\strtotime;
+
+/**
+ * Building a Redia-legacy RSS feed, showing eventinstances.
+ */
+class EventsController extends ControllerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    protected FileUrlGeneratorInterface $fileUrlGenerator,
+  ) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): static {
+    return new static(
+      $container->get('file_url_generator')
+
+    );
+  }
+
+  /**
+   * Getting the RSS/XML feed of the items.
+   */
+  public function getFeed(Request $request): CacheableResponse {
+    $items = $this->getItems();
+
+    $rss_content = $this->buildRss($items, $request);
+
+    $response = new CacheableResponse();
+    $response->setContent($rss_content);
+    $response->headers->set('Content-Type', 'application/rss+xml');
+    return $response;
+  }
+
+  /**
+   * Loading events, and turning it into a simple array of relevant values.
+   *
+   * @return \Drupal\dpl_redia_legacy\RediaEvent[]
+   *   An array of necessary item fields, used in buildRss().
+   */
+  private function getItems(): array {
+
+    $storage = $this->entityTypeManager()->getStorage('eventinstance');
+    $query = $storage->getQuery()
+      ->condition('status', TRUE)
+      ->condition('date.value', strtotime('today'), '>=')
+      ->accessCheck(TRUE)
+      ->sort('date.value');
+    $ids = $query->execute();
+
+    /** @var \Drupal\recurring_events\Entity\EventInstance[] $events */
+    $events = $this->entityTypeManager()->getStorage('eventinstance')->loadMultiple($ids);
```

Comment:
```suggestion
    $events = $storage->loadMultiple($ids);
```

As I see it we already have the storage available. Is there a reason to retrieve it again?